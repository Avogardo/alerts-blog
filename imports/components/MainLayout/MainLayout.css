.content-container {
	width: auto;
	background-color: #ffffff;
	margin: 0 15px;
}
