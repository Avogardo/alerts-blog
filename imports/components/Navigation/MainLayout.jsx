import React from 'react';

const Navigation = () => (
  <nav>
    Navbar
  </nav>
);

export default Navigation;
