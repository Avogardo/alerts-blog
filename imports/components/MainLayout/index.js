import { Meteor } from 'meteor/meteor';
import { Tracker } from 'meteor/tracker';
import { compose } from 'react-komposer';
import {
  isLoggedIn,
  onLogOut,
  isAdmin,
  actions as userActions,
} from '../../api/users';
import { actions as newsActions } from '../../api/news';
import MainLayout from './MainLayout.jsx';

const getTrackerLoader = composer =>
  (props, onData, env) => {
    let trackerCleanup = null;
    const handler = Tracker.nonreactive(() =>
      Tracker.autorun(() => {
        // assign the custom clean-up function.
        trackerCleanup = composer(props, onData, env);
      }));

    return () => {
      if (typeof trackerCleanup === 'function') trackerCleanup();
      return handler.stop();
    };
  };


const composer = (props, onData) => {
  const userHandler = Meteor.subscribe('extendedUser');
  const { goToSignIn } = userActions;
  const { goToNewsContainer, goToCreateNews } = newsActions;
  const isLoggedInUser = isLoggedIn();
  const isAuthorized = isAdmin();

  if (userHandler.ready()) {
    onData(null, {
      goToSignIn,
      goToNewsContainer,
      goToCreateNews,
      isLoggedInUser,
      onLogOut,
      isAuthorized,
      ...props,
    });
  }
};

export default compose(getTrackerLoader(composer))(MainLayout);
