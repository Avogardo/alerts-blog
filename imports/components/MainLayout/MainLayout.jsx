import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Route, Redirect } from 'react-router-dom';
import ReactRouterPropTypes from 'react-router-prop-types';

import Navigation from '../Navigation';
import Sidebar from '../Sidebar';
import NewsContainer from '../NewsContainer';
import SignIn from '../SignIn';
import CreateNews from '../CreateNews';
import './MainLayout.css';

class MainLayout extends Component {
  constructor(props) {
    super(props);
    this.toggleSidebar = this.toggleSidebar.bind(this);
    this.goToSignIn = this.goToSignIn.bind(this);
    this.goToNewsContainer = this.goToNewsContainer.bind(this);
    this.goToCreateNews = this.goToCreateNews.bind(this);
    this.onLogOut = this.onLogOut.bind(this);

    this.state = {
      isSidebarOpen: false,
    };
  }

  onLogOut() {
    const { onLogOut, history } = this.props;

    this.toggleSidebar();
    onLogOut(history);
  }

  setSidebarState(open) {
    this.setState({
      isSidebarOpen: open,
    });
  }

  toggleSidebar() {
    this.setState({
      isSidebarOpen: !this.state.isSidebarOpen,
    });
  }

  goToSignIn() {
    const { goToSignIn, history } = this.props;

    this.toggleSidebar();
    goToSignIn(history);
  }

  goToNewsContainer() {
    const { goToNewsContainer, history } = this.props;
    goToNewsContainer(history);
  }

  goToCreateNews() {
    const { goToCreateNews, history } = this.props;

    this.toggleSidebar();
    goToCreateNews(history);
  }

  render() {
    const { isSidebarOpen } = this.state;
    const { isLoggedInUser, isAuthorized, location } = this.props;

    return (
      <div className="container">
        <Navigation
          toggleSidebar={this.toggleSidebar}
          goToNewsContainer={this.goToNewsContainer}
        />
        <Sidebar
          isSidebarOpen={isSidebarOpen}
          setSidebarState={open => this.setSidebarState(open)}
          goToSignIn={this.goToSignIn}
          goToCreateNews={this.goToCreateNews}
          onLogOut={this.onLogOut}
          isAuthorized={isAuthorized}
        />
        <div
          className={location.pathname === '/' ? 'content-container-root' : 'content-container'}
          key="content-container"
        >
          <Route
            exact
            path="/"
            render={() => (
              <Fragment>
                <NewsContainer enterContainer />
                <NewsContainer headerTitle="Latest News" />
                <NewsContainer exitNews headerTitle="Editor`s Pick" />
              </Fragment>
            )}
          />
          <Route
            exact
            path="/sign-in"
            render={props => (
              isLoggedInUser
                ? <Redirect to="/" />
                : <SignIn {...props} />
              )}
          />
          <Route
            exact
            path="/create-news"
            render={() => (
              isAuthorized
                ? <CreateNews />
                : <Redirect to="/" />
            )}
          />
        </div>
      </div>
    );
  }
}

MainLayout.propTypes = {
  goToSignIn: PropTypes.func.isRequired,
  goToNewsContainer: PropTypes.func.isRequired,
  goToCreateNews: PropTypes.func.isRequired,
  onLogOut: PropTypes.func.isRequired,
  history: ReactRouterPropTypes.history.isRequired,
  isLoggedInUser: PropTypes.bool.isRequired,
  isAuthorized: PropTypes.bool.isRequired,
  location: ReactRouterPropTypes.location.isRequired,
};

export default MainLayout;
