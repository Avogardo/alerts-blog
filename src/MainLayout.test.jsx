import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import MainLayout from '../imports/components/MainLayout';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MainLayout />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly all elements', () => {
  const tree = TestRenderer.create(<MainLayout />).toJSON();
  expect(tree).toMatchSnapshot();
});
