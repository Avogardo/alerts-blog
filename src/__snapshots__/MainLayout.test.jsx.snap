// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly all elements 1`] = `
<div>
  <header
    className="MuiPaper-root-92 MuiPaper-elevation4-98 MuiAppBar-root-84 MuiAppBar-positionStatic-88 MuiAppBar-colorPrimary-90"
    style={
      Object {
        "backgroundColor": "#04091e",
      }
    }
  >
    <div
      className="MuiToolbar-root-119 MuiToolbar-gutters-120"
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <div>
        <button
          className="MuiButtonBase-root-127 MuiIconButton-root-121 MuiIconButton-colorInherit-122"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-126"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M17,2V2H17V6H15C14.31,6 14,6.81 14,7.5V10H14L17,10V14H14V22H10V14H7V10H10V6A4,4 0 0,1 14,2H17Z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-130"
          />
        </button>
        <button
          className="MuiButtonBase-root-127 MuiIconButton-root-121 MuiIconButton-colorInherit-122"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-126"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M10,15L15.19,12L10,9V15M21.56,7.17C21.69,7.64 21.78,8.27 21.84,9.07C21.91,
 9.87 21.94,10.56 21.94,11.16L22,12C22,14.19 21.84,15.8 21.56,16.83C21.31,
 17.73 20.73,18.31 19.83,18.56C19.36,18.69 18.5,18.78 17.18,18.84C15.88,
 18.91 14.69,18.94 13.59,18.94L12,19C7.81,19 5.2,18.84 4.17,18.56C3.27,
 18.31 2.69,17.73 2.44,16.83C2.31,16.36 2.22,15.73 2.16,14.93C2.09,14.13 2.06,
 13.44 2.06,12.84L2,12C2,9.81 2.16,8.2 2.44,7.17C2.69,6.27 3.27,5.69 4.17,
 5.44C4.64,5.31 5.5,5.22 6.82,5.16C8.12,5.09 9.31,5.06 10.41,5.06L12,5C16.19,
 5 18.8,5.16 19.83,5.44C20.73,5.69 21.31,6.27 21.56,7.17Z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-130"
          />
        </button>
      </div>
      <button
        className="MuiButtonBase-root-127 MuiIconButton-root-121 MuiIconButton-colorInherit-122"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-126"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M4,4H20A2,2 0 0,1 22,6V18A2,2 0 0,1 20,20H4C2.89,20 2,19.1 2,18V6C2,4.89 2.89,
 4 4,4M12,11L20,6H4L12,11M4,18H20V8.37L12,13.36L4,8.37V18Z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-130"
        />
      </button>
    </div>
  </header>
  <div
    className="logo-container"
  >
    <img
      alt="logo"
      src="https://drive.google.com/uc?id=1U_HEoR8c2kubf6-JsbHEcwo564J5zjlE"
    />
    <h1>
      Alarms blog
    </h1>
  </div>
  <header
    className="MuiPaper-root-92 MuiPaper-elevation4-98 MuiAppBar-root-84 MuiAppBar-positionSticky-87 MuiAppBar-colorPrimary-90"
    style={
      Object {
        "backgroundColor": "#04091e",
        "margin": "0 15px",
        "width": "auto",
      }
    }
  >
    <div
      className="MuiToolbar-root-119 MuiToolbar-gutters-120"
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <div
        className="navigation-wrapper"
      >
        <button
          aria-label="Menu"
          className="MuiButtonBase-root-127 MuiIconButton-root-121 MuiIconButton-colorInherit-122"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-126"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-137"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g>
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </g>
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-130"
          />
        </button>
        <h2
          className="MuiTypography-root-143 MuiTypography-title-149 MuiTypography-colorInherit-162"
        >
          Menu
        </h2>
      </div>
      <button
        className="MuiButtonBase-root-127 MuiIconButton-root-121 MuiIconButton-colorInherit-122"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-126"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-137"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-130"
        />
      </button>
    </div>
  </header>
</div>
`;
